src/components/pages/farmer-produce/list/index.tsx

```diff
@@ -108,6 +137,15 @@ function FarmerListPageComponent() {
         >
           Create Batch
         </Button>
+        <Button
+          colorScheme="gray"
+          variant="solid"
+          onClick={handleSyncData}
+          isDisabled={showTypeError || isSyncing || !hasAccess([ROLES.ADMIN, ROLES.UNION], user)}
+          leftIcon={isSyncing ? <Spinner size="xs" /> : <RepeatIcon />}
+        >
+          {isSyncing ? "Syncing..." : "Sync Data"}
```

Comment:
like in my previous comment. if we put this text in translation file, we can resuse it here.